arrows/ffmpeg/ffmpeg_video_input.cxx

```diff
@@ -1965,9 +2217,21 @@ ::timestamp() const
   {
     return {};
   }
-  return kv::timestamp{
-    static_cast< kv::time_usec_t >( curr_time() * 1000000.0 + 0.5 ),
-    frame_number() + 1 };
+
+  kv::timestamp ts;
```

Comment:
I assume this can be rewritten in the following way
```
  if( !frame.has_value() )
  {
    return {};
  }

  kv::timestamp ts;
    if( frame_rate().num > 0 )
  {
    ts.set_frame( frame_number() + 1 );
  } else if( frame->frame->best_effort_timestamp != AV_NOPTS_VALUE )
  {
    ts.set_time_usec(
      static_cast< kv::time_usec_t >( curr_time() * 1000000.0 + 0.5 ) );
  }
  return ts;
```
with this we don't need to check frame.has_value() twice (because initially its just returning with `!frame.has_value()`. Also avoid two independent if blocks, instead on if-else is enough.

Reply:
Changing it to an `else if` is not correct, because that changes the logic - in most cases we want both blocks to run. However, you are correct that the second `frame.has_value()` check is redundant.

Reply:
Agree, my fault, somehow I had an impression like in two blocks its updating the same value.